Monitor tempdb - query change

scott-epperly opened the pull request:
The tempdb monitor query does not appear to return appropriate results for the following reasons:
1. The sql_requests view returns historical values from sys.dm_pdw_sql_requests.  These inevitably join to the node level results which are real-time only.  This can be confusing to the user as unintentional joins on historical, reused spids are surfaced in the results.
2. The sql_requests view only takes sys.dm_pdw_sql_requests into account; however, tempdb usage from the sister view, sys.dm_pdw_dms_workers could be reported on as well.
3. @@SPID does not return a DW session id; use the SESSION_ID() function to filter out the monitoring query.  However, I don't believe this filter is necessary at all as the joins to the node level DMVs will eliminate the current query.

The suggested change implements the following:
1. Removes the view and uses a CTE instead; this is to prevent creation of a permanent object in a user database_id
2. Adds sys.dm_pdw_dms_workers rows to the reported data set
3. Filters for only currently running queries
4. Removes unnecessary filters for current session and 'sa' login
5. Orders by request_id and pdw_node_id
6. Provides an inline example for filtering against a particular session_id
7. Removed verbiage relating to the view creation
8. Added verbiage indicating that this query can only be used for real-time monitoring only

I expect this change to be reviewed pretty closely since there's a suggested code change.  Please note that there is an assumption built into the CTE that all final statuses in sys.dm_pdw_sql_requests begin with "StepC" or "StepE" - this assumption may not be accurate, but documentation on the DMV doesn't provide enough context to determine all possible final statuses.

azurecla commented:
Hi __@sahaj08__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (sahajs). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

rmca14 commented:
@JennieHubbard 
Please review and approve here, thanks.
@barbkess 
FYI

rmca14 commented:
@JennieHubbard 
Jennie, following up in case you missed this mail. Thanks.

JennieHubbard commented:
#sign-off

scott-epperly committed:
Monitor tempdb - query change

The tempdb monitor query does not appear to return appropriate results for the following reasons:
1. The sql_requests view returns historical values from sys.dm_pdw_sql_requests.  These inevitably join to the node level results which are real-time only.  This can be confusing to the user as unintentional joins on historical, reused spids are surfaced in the results.
2. The sql_requests view only takes sys.dm_pdw_sql_requests into account; however, tempdb usage from the sister view, sys.dm_pdw_dms_workers could be reported on as well.
3. @@SPID does not return a DW session id; use the SESSION_ID() function to filter out the monitoring query.  However, I don't believe this filter is necessary at all as the joins to the node level DMVs will eliminate the current query.

The suggested change implements the following:
1. Removes the view and uses a CTE instead; this is to prevent creation of a permanent object in a user database_id
2. Adds sys.dm_pdw_dms_workers rows to the reported data set
3. Filters for only currently running queries
4. Removes unnecessary filters for current session and 'sa' login
5. Orders by request_id and pdw_node_id
6. Provides an inline example for filtering against a particular session_id
7. Removed verbiage relating to the view creation
8. Added verbiage indicating that this query can only be used for real-time monitoring only

I expect this change to be reviewed pretty closely since there's a suggested code change.  Please note that there is an assumption built into the CTE that all final statuses in sys.dm_pdw_sql_requests begin with "StepC" or "StepE" - this assumption may not be accurate, but documentation on the DMV doesn't provide enough context to determine all possible final statuses.
